MASFoundation/Classes/models/Network/MASRequestBuilder.h

```diff
@@ -0,0 +1,174 @@
+//
+//  MASRequestBuilder.h
+//  MASFoundation
+//
+//  Copyright (c) 2017 CA. All rights reserved.
+//
+//  This software may be modified and distributed under the terms
+//  of the MIT license. See the LICENSE file for details.
+//
+
+#import <Foundation/Foundation.h>
+
+#import "MASClaims.h"
+
+/**
+ MASRequestBuilder class is an object that allows developers to progressively build a request as needed.
+ The class is mainly responsible for receive parameters and create a MASRequest object.
+ 
+ Default configuration value for designated initializer, [[MASRequestBuilder alloc] initWithHTTPMethod:], would be:
+ isPublic: NO,
+ sign: NO,
+ requestType:MASRequestResponseTypeJson, 
+ responseType:MASRequestResponseTypeJson.
+ */
+@interface MASRequestBuilder : NSObject
+
+NS_ASSUME_NONNULL_BEGIN
+
+///--------------------------------------
+/// @name Properties
+///--------------------------------------
+
+# pragma mark - Properties
+
+
+/**
+ NSString value of the HTTP Method (GET, POST, PUT, DELETE).
+ */
+@property (nonatomic, strong, readonly) NSString *httpMethod;
+
+/**
+ BOOL value that determines whether or not the target host is a primary gateway or another gateway/public server.
```

Comment:
BOOL value that determines whether or not to include credentials of primary gateway in the request.